packages/web/pages/transactions.tsx

```diff
@@ -38,10 +49,11 @@ const Transactions: React.FC = observer(() => {
   const { data: transactionData, isLoading } =
     api.edge.transactions.getTransactions.useQuery(
```

Comment:
Is there a reason you're not using `useInfiniteQuery`? We have some server side utils that make paginating over an array of items easy, and it cooperates will with the pagination tooling within React Query. See asset-info and asset-balances pages for examples (along with their tRPC procedures)

Reply:
Like here, all you need to do is wrap your returned array in `maybeCachePaginatedItems`
https://github.com/osmosis-labs/osmosis-frontend/blob/eb445804f479847c2fd99639f00233aa5a3389a8/packages/server/src/trpc-routers/assets-router.ts#L84